### PR TITLE
Adds customization option for Executors coming from AsyncConfigurers

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
Related to https://github.com/spring-projects/spring-boot/issues/29151
Required by https://github.com/spring-cloud/spring-cloud-sleuth/issues/2100

cc @snicoll @wilkinsona

### CODE REVIEW

File: spring-context/src/main/java/org/springframework/scheduling/annotation/AbstractAsyncConfiguration.java [hunk 1]
```diff
@@ -81,7 +81,8 @@ void setConfigurers(ObjectProvider<AsyncConfigurer> configurers) {
 			}
 			return candidates.get(0);
 		});
-		this.executor = adapt(configurer, AsyncConfigurer::getAsyncExecutor);
+		Supplier<Executor> adaptedExecutor = adapt(configurer, AsyncConfigurer::getAsyncExecutor);
+		customizers.ifAvailable(customizer -> this.executor = () -> customizer.customize(adaptedExecutor.get()));
```
Comment (nkonev) [inline spring-context/src/main/java/org/springframework/scheduling/annotation/AbstractAsyncConfiguration.java:85]: `1. this.executor` won't be set if customizers aren't available. Is this desired ?
2. customizers has plural form but in fact it's a single customizer, not a collection

Comment (marcingrzejszczak) [reply spring-context/src/main/java/org/springframework/scheduling/annotation/AbstractAsyncConfiguration.java:85]: Nope, good catch

**Review by nkonev — APPROVED** [review]

**Comment by maradanasai** [comment]
This is happening for us when we are migrating to latest. Here is the sample reproducible code https://github.com/maradanasai/otel-async-exper.git . Using spring boot 2.7.14 and spring cloud 2021.0.8 version with jdk 17. It is blocking migrating to latest versions. Please fix this

**Comment by snicoll** [comment]
Sorry for the delay. Unfortunately, we can't add such an interface in 5.3.x at this time, in particular as it is not required in the next generation of micrometer tracing and the framework. I think Sleuth should rather reintroduce the BPP that used to swap the executor, or document how it can be done. This also brings more consistency with what we have to do in the next major release.

**Pull request closed without merging**